### ISSUE TITLE
unnamedResult: checkExported does not check unexported anymore

### ISSUE THREAD

**Issue by hu3bi** [go-critic/go-critic#1524]
https://go-critic.com/overview.html#unnamedresult

What should happen:
With
```go
        unnamedResult:
          # Whether to check exported functions.
          # Default: false
          checkExported: true
```
It should check all exported and unexported functions.

What actually happened:
It only checked exported functions and ignored unexported.

---

How to fix:
a) Either rename `checkExported` to `checkExportedOnly` or `skipUnexported`.
b) or implement a behavior that represents the rule.

**Comment by cristaloleg** [comment]
Hey, do you have a reproducer?

**Comment by hu3bi** [comment]
See the different output below:

<img width="2237" height="1502" alt="Image" src="https://github.com/user-attachments/assets/4e07cff8-5d35-4e55-858b-f106d3759933" />
<img width="2262" height="1410" alt="Image" src="https://github.com/user-attachments/assets/95dce99f-47df-44a1-9582-8dc0b1edc55f" />

code:
https://go.dev/play/p/JyymX9xEGqd
